File: app/src/main/java/com/nvvi9/giphyme/domain/mappers/GifDetailsItemMapper.java
package com.nvvi9.giphyme.domain.mappers;

import com.nvvi9.giphyme.data.BaseMapper;
import com.nvvi9.giphyme.data.gifinfo.GifInfo;
import com.nvvi9.giphyme.vo.GifDetailsItem;

public class GifDetailsItemMapper implements BaseMapper<GifInfo, GifDetailsItem> {

    @Override
    public GifDetailsItem map(GifInfo value) {
        GifDetailsItem.GifAuthorDetailsItem gifAuthorDetailsItem = new GifDetailsItem.GifAuthorDetailsItem(value.getAuthorName(), value.getAuthorAvatarUrl(), value.getAuthorProfileUrl());
        return new GifDetailsItem(value.getId(), value.getTitle(), value.getGifUrl(), value.getUrl(), gifAuthorDetailsItem, value.getCreatedAt());
    }
}

File: app/src/main/java/com/nvvi9/giphyme/domain/mappers/GifDetailsItemMapper.kt
package com.nvvi9.giphyme.domain.mappers

import com.nvvi9.giphyme.data.BaseMapper
import com.nvvi9.giphyme.data.gifinfo.GifInfo
import com.nvvi9.giphyme.vo.GifDetailsItem
import com.nvvi9.giphyme.vo.GifDetailsItem.GifAuthorDetailsItem

class GifDetailsItemMapper : BaseMapper<GifInfo, GifDetailsItem> {

    override fun map(value: GifInfo): GifDetailsItem =
        GifDetailsItem(
            value.id,
            value.title,
            value.gifUrl,
            value.url,
            GifAuthorDetailsItem(value.authorName, value.authorAvatarUrl, value.authorProfileUrl),
            value.createdAt
        )
}